### PR TITLE
clarify type requirements for search intersects parameter, update some documentation

### PR DESCRIPTION
**Pull request by philvarner**
**Related Issue(s):** #146 

**Description:**

1. For search intersects, tighten type definitions to require str, dict, or object with __geo_interface__ attribute. Other objects now fail. If a user wants to use another object, they have to str() it first.
2. add better documentation about this.

**PR Checklist:**

- [X] Code is formatted
- [X] Tests pass
- [X] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac-api-client/blob/main/CHANGELOG.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-STAC Client
-===============
+# STAC Client <!-- omit in toc -->
```
Comment (philvarner) [inline README.md:1]: this is so it won't show up in the TOC when autogenerated by VSCode All-in-one

File: tests/test_item_search.py [hunk 2]
```diff
@@ -377,7 +381,9 @@ def test_intersects_results(self):
 
         # Geo-interface object
         class MockGeoObject:
-            __geo_interface__ = intersects_dict
```
Comment (philvarner) [inline tests/test_item_search.py:380]: I don't think this actually tested what it was intended to test before. Anyway, I fixed it so the typechecking works now anyway

File: pystac_client/item_search.py [hunk 3]
```diff
@@ -409,9 +417,15 @@ def _format_fields(self, value: Optional[FieldsLike]) -> Optional[Fields]:
     def _format_intersects(value: Optional[IntersectsLike]) -> Optional[Intersects]:
         if value is None:
             return None
+        if isinstance(value, dict):
+            return deepcopy(value)
         if isinstance(value, str):
             return json.loads(value)
-        return deepcopy(getattr(value, '__geo_interface__', value))
+        if hasattr(value, '__geo_interface__'):
```
Comment (gadomski) [inline pystac_client/item_search.py:424]: Can you add a comment to add [runtime_checkable](https://docs.python.org/3/library/typing.html#typing.runtime_checkable) when we drop 3.7 support?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pystac_client/item_search.py (hunk 3 above)
Comment (philvarner) [reply pystac_client/item_search.py:425]: I didn't know about that before, thanks

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pystac_client/item_search.py [hunk 4]
```diff
@@ -22,6 +22,13 @@
                             r"(?P<remainder>(T|t)\d{2}:\d{2}:\d{2}(\.\d+)?"
                             r"(?P<tz_info>Z|([-+])(\d{2}):(\d{2}))?)?)?)?")
 
+
+# todo: add runtime_checkable when
```
[the remark this answers is not in the document]
Comment (philvarner) [reply pystac_client/item_search.py:26]: when we lose track of what we're writing

**Review by gadomski — APPROVED** [review]

**Pull request merged**